vsphere/resource_vsphere_host.go

```diff
@@ -817,3 +912,206 @@ func (h HostAccessManager) ChangeLockdownMode(ctx context.Context, mode types.Ho
 	_, err := methods.ChangeLockdownMode(ctx, h.Client(), &req)
 	return err
 }
+
+func resourceVSphereHostUpdateServices(d *schema.ResourceData, meta interface{}, oldVal, newVal interface{}) error {
+	client := meta.(*Client).vimClient
+	hostID := d.Id()
+	hostObject, err := hostsystem.FromID(client, hostID)
+	if err != nil {
+		return fmt.Errorf("error while retrieving HostSystem object for host ID %s. Error: %s", hostID, err)
+	}
+
+	servicesSet, ok := d.Get("services").(*schema.Set)
+	if !ok {
+		return fmt.Errorf("error reading 'services': type assertion to *schema.Set failed")
+	}
+
+	updatedServices := make([]interface{}, 0) // Prepare to collect updated services configurations
+
+	services := servicesSet.List()
+	for _, service := range services {
+		serviceMap := service.(map[string]interface{})
+		updatedServiceMap := make(map[string]interface{}) // Prepare to collect updated service configuration
+
+		if ntpd, ok := serviceMap["ntpd"]; ok {
```

Comment:
I have a general question about the design of this method.

It appears to be intended to be generic, at least that's what the name suggests. I thought that was the idea behing the `mutableKeys` map on line 316.
In the latest revision this method has specific handling for `ntpd`. I think you should decide which direction you want to go in - a generic method and `mutableKeys` or no `mutableKeys` and a method with custom handling for each service (preferably with submethods for individual services)

Reply:
i am just playing this is not final

Reply:
So for testing not using mutableKeys for the actually modification using mutable keys still.